### PR TITLE
Removed credentials file

### PR DESCRIPTION
**Pull request by hvsw**
Removed the `Credentials.plist` file and save the credentials on user's Keychain.

## Benefits
- Since is not necessary to set credentials on source files and rebuild, users will be able to download unsigned version and use authenticated gists.
- Users won't need to worry with version control since the credentials are on the user's Keychain.

### CODE REVIEW

File: macGist/GitHubCredential.swift [hunk 1]
```diff
@@ -8,21 +8,48 @@
...
     
-    var value: String {
-        guard let path = Bundle.main.path(forResource: "Credentials", ofType: "plist"),
-            let dict = NSDictionary(contentsOfFile: path) as? [String: AnyObject] else { fatalError("No plist found") }
+    static var clientId: String? {
+        get {
+            do {
+                let clientId = try keychain.readGitHubClientId()
+                return clientId
+            } catch {
```
Comment (unnamedd) [inline macGist/GitHubCredential.swift]: if you aren't really using this catch, will not be better if you use:
```swift
return try? keychain.ReadGitHubClientId()
```
?

File: macGist/GitHubCredential.swift [hunk 2]
```diff
@@ -8,21 +8,48 @@
...
+            }
+        }
+    }
+    
+    static var clientSecret: String? {
+        get {
+            do {
+                let clientSecret = try keychain.readGitHubClientSecret()
+                return clientSecret
+            } catch {
```
Comment (unnamedd) [inline macGist/GitHubCredential.swift]: take a look in this [comment](https://github.com/Bunn/macGist/pull/11/files/9d1102f807c8fc259e10fc34ea24183f1a88e493#diff-3710e79a4e3339cf0255b43d88314eb8)

File: macGist/Keychain.swift [hunk 3]
```diff
@@ -7,6 +7,59 @@ struct KeychainConfiguration {
...
+        
+        static func query() -> [String:AnyObject] {
+            let query = KeychainConfiguration.keychainQuery(withService: KeychainConfiguration.GitHub.serviceName,
+                                                     account: KeychainConfiguration.GitHub.account,
+                                                     accessGroup: KeychainConfiguration.accessGroup)
+            return query
+        }
+    }
+    
+    static func query() -> [String:AnyObject] {
```
Comment (unnamedd) [inline macGist/Keychain.swift]: Why do we have two implementations of this method?

File: macGist/Keychain.swift [hunk 4]
```diff
@@ -84,15 +123,124 @@ struct Keychain {
...
         
-        // Throw an error if an unexpected status was returned.
         guard status == noErr || status == errSecItemNotFound else { throw KeychainError.unhandledError(status: status) }
     }
     
+    func save(gitHubClientId: String) throws {
+        let encodedGitHubClientId = gitHubClientId.data(using: String.Encoding.utf8)!
+        
+        do {
+            try _ = readGitHubClientId()
```
Comment (unnamedd) [inline macGist/Keychain.swift]:
```swift
try readGitHubClientId()
```
it isn't better?

File: macGist/Keychain.swift [hunk 5]
```diff
@@ -16,64 +69,50 @@ struct Keychain {
...
     
-    func savePassword(_ password: String) throws {
-        // Encode the password into an Data object.
+    func save(token password: String) throws {
         let encodedPassword = password.data(using: String.Encoding.utf8)!
         
         do {
-            // Check for an existing item in the keychain.
-            try _ = readPassword()
+            try _ = readToken()
```
Comment (unnamedd) [inline macGist/Keychain.swift]:
```swift
try readToken()
```
it isn't better?

File: macGist/Keychain.swift [hunk 6]
```diff
@@ -84,15 +123,124 @@ struct Keychain {
...
+            let status = SecItemAdd(newItem as CFDictionary, nil)
+            
+            guard status == noErr else { throw KeychainError.unhandledError(status: status) }
+        }
+    }
+    
+    func readGitHubClientId() throws -> String {
+        var query = KeychainConfiguration.GitHub.Gist.Account.clientId.query()
+        
+        query[kSecMatchLimit as String] = kSecMatchLimitOne
```
Comment (unnamedd) [inline macGist/Keychain.swift]: to be honest, when I see `as String` a lot of times, I feel not comfortable... 😐

**Review by unnamedd — CHANGES_REQUESTED** [review]

File: macGist/Keychain.swift (hunk 4 above)
Comment (hvsw) [reply macGist/Keychain.swift]: Adjusted and added `@discardableResult` to suppress the warning.

File: macGist/Keychain.swift (hunk 3 above)
Comment (hvsw) [reply macGist/Keychain.swift]: Removed implementation from `Keychain`.

File: macGist/GitHubCredential.swift (hunk 1 above)
Comment (hvsw) [reply macGist/GitHubCredential.swift]: Suggestion accepted.

File: macGist/GitHubCredential.swift (hunk 2 above)
Comment (hvsw) [reply macGist/GitHubCredential.swift]: Suggestion accepted.

File: macGist/Keychain.swift (hunk 5 above)
Comment (hvsw) [reply macGist/Keychain.swift]: Suggestion accepted.

File: macGist/Keychain.swift (hunk 6 above)
Comment (hvsw) [reply macGist/Keychain.swift]: Removed a lot of casts. Created a typealias for `[CFString:AnyObject]` so now the query will be of this type.

**Review by unnamedd — APPROVED** [review]

**Comment by Bunn** [comment]
@hvsw @unnamedd I'm not really sure if I understood the point of moving the credentials to the keychain.

This is not the user's account credential but the app's GitHub key data, which needs to be embedded when creating the binary version. This won't work with a keychain version because every user that downloads the binary will need to create a GitHub App and get the application key, which shouldn't be required for binary versions, only when compiled from source.

The actual user data was already being stored on the keychain, the plist was just for GitHub App Key.

**Pull request merged**